detox/src/realms/DetoxPrimaryContext.test.js

```diff
@@ -1,62 +1,73 @@
 
 describe('DetoxPrimaryContext', () => {
+  //#region Fixtures and constants
+  const UUID_REGEXP_STR = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}';
+  const UUID_REGEXP = new RegExp(`^${UUID_REGEXP_STR}$`);
+  const TEMP_FILE_REGEXP = new RegExp(`.*${UUID_REGEXP_STR}.detox.json$`);
+  const WORKER_ID = 19;
+  const DETOX_CONFIG_BASE = Object.freeze({
+    behavior: {
+      init: {
+        keepLockFile: false,
+      }
+    },
+    device: {
+      type: '',
+    },
+    logger: {},
+    session: {
+      autoStart: false,
+    },
+  });
+  const FIRST_ARGUMENT = 0;
+  //#endregion
+
   let detoxConfigDriver;
 
+  //#region Mocks
+  /** @type {jest.Mocked<import('fs-extra')>} */
   let fs;
-  /** @type {import('../utils/logger')} */
+  /** @type {jest.Mocked<import('signal-exit')>} */
+  let signalExit;
+  /** @type {jest.Mocked<import('../logger')>} */
   let logger;
-
-  /** @type {import('../configuration')} */
+  /** @type {jest.Mocked<import('../configuration')>} */
   let configuration;
-  /** @type {import('../ipc/IPCServer')} */
+  /** @type {jest.Mock<import('../ipc/IPCServer')>} */
   let IPCServer;
-  /** @type {import('../ipc/IPCServer')} */
-  let ipcServer;
-
+  /** @type {jest.Mocked<import('../devices/lifecycle/GenyGlobalLifecycleHandler')>} */
   let globalLifecycleHandler;
-  /** @type {import('../environmentFactory')} */
+  /** @type {jest.Mocked<import('../environmentFactory')>} */
   let environmentFactory;
-  /** @type {import('../devices/DeviceRegistry')} */
+  /** @type {jest.Mocked<import('../devices/DeviceRegistry')>} */
   let deviceRegistryIOS;
-  /** @type {import('../devices/DeviceRegistry')} */
+  /** @type {jest.Mocked<import('../devices/DeviceRegistry')>} */
   let deviceRegistryAndroid;
-  /** @type {import('../devices/DeviceRegistry')} */
+  /** @type {jest.Mocked<import('../devices/DeviceRegistry')>} */
   let deviceRegistryGenyCloud;
-  /** @type {import('../server/DetoxServer')} */
+  /** @type {jest.Mock<import('../server/DetoxServer')>} */
   let DetoxServer;
+  /** @type {jest.Mock<import('../DetoxWorker')>} */
+  let DetoxWorker;
+  //#endregion
 
   /** @type {import('./DetoxInternalsFacade')} */
   let facade;
 
-  let env;
-
   const detoxServer = () => latestInstanceOf(DetoxServer);
-  const detoxWorker = () => {
-    const DetoxWorker = require('../DetoxWorker');
-    return latestInstanceOf(DetoxWorker);
-  };
+  const ipcServer = () => latestInstanceOf(IPCServer);
+  const detoxWorker = () => latestInstanceOf(DetoxWorker);
   const logFinalizer = () => latestInstanceOf(logger.DetoxLogFinalizer);
-  const getSignalHandler = () => {
-    const onSignalExit = require('signal-exit');
-    const [signalHandler] = lastCallTo(onSignalExit);
-    return signalHandler;
-  };
+  const getSignalHandler = () => lastCallTo(signalExit)[FIRST_ARGUMENT];
 
-  beforeEach(() => env = process.env);
-  afterEach(() => process.env = { ...env });
+  backupProcessEnv();
```

Comment:
Looks like we could use this helper in much more suites, and eventually, we might do that.

Reply:
Might not be a bad idea to register global before-after callbacks that do this for all tests

Reply:
Well, maybe, but it's rather a red flag for me. It will worsen our self-tracking of code that depends on `process.env`. It took me a few years to get rid of `process.env` dependencies here and there (`argparse` util counts in too, because It does the same under the hood).

It would be convenient to have it explicit so we can search in the project and know exactly how many modules-under-the-test have this nasty tendency to poke in `process.env`. I'd rather had this as an exception to the rule than make every module/unit test so careless.